teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,326 @@
+
+```yaml
+taskRef:
+  resolver: bundles
+  name: "oci://10.96.190.208:5000/ptasci67/example-oci@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828?kind=task&name=hello-world?secret=default"
+```
+
+##### Http Resolver
+Based on `http` resolver's [params](https://tekton.dev/docs/pipelines/http-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+<http-url>?http-username=<http-username>&http-password-secret=<http-password-secret>&http-password-secret-key=<http-password-secret-key>
+```
+
+Here, the url contains the `scheme` and `location`: `<scheme>://<locaiton>`:
+
+Example:
+```
+http://example.com/foo/task.yaml
+https://raw.githubusercontent.com/tektoncd-catalog/git-clone/main/task/git-clone/git-clone.yaml
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  resolver: http
+  name: "https://raw.githubusercontent.com/tektoncd-catalog/git-clone/main/task/git-clone/git-clone.yaml?http-username=git&http-password-secret=git-secret&http-password-secret-key=git-token"
+```
+
+##### Hub Resolver
+Based on `hub` resolver's [params](https://tekton.dev/docs/pipelines/hub-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+artifacthub://<catalog>/<kind>/<name>@<version>
+tektonhub://<catalog>/<kind>/<name>@<version>
+```
+**Note** that we use a scheme called `artifacthub` or `tektonhub` which is only meaningful to the hub resolver.
+
+Example:
+
+```
+artifacthub://tekton-catalog-tasks/task/git-clone@0.6
+```
+
+Using the above example, a user can do the following when referencing the Task:
+
+```yaml
+taskRef:
+  resolver: hub
+  name: "artifacthub://tekton-catalog-tasks/task/git-clone@0.6"
+```
+
+##### Cluster Resolver
+Based on `cluster` resolver's [params](https://tekton.dev/docs/pipelines/cluster-resolver/#parameters), we define the following `resolver-reference` format: 
+
+```
+cluster://<namespace>/<kind>/<name>
+```
+
+**Note** that we use a scheme called `cluster` which is only meaningful to the cluster resolver.
+
+Example:
+
+```
+cluster://default/task/git-clone
+```
+
+Using the above example, a user can do the following when referencing the Task:
+
+```yaml
+taskRef:
+  resolver: "cluster"
+  name: "cluster://default/task/git-clone" 
+```
+
+##### Custom Resolver
+When implementing a new custom resolver, just like users have the freedom to choose their parameters, they also have the freedom to define their `name` (or `url`) based on:
+
+```
+<scheme>://<location>[@<version>][#<selector>][?<params>]
+```
+
+The custom resolver can choose the schemes it wants to subscribe to and for each scheme, it can define the remainder of `location, version, selectors and params` as needed. An example `scheme` could be `<custom-resolver-name>`. A `location` may be made up of concrete units e.g. `location = <foo>/<bar>` (i.e. expect two parts) that are up to the discretion of the custom resolvers.
```

Comment:
The subscribe part applies for stage 2 only though, right ?

Reply:
Yes, let me clarify that.

Reply:
done.